File: apps/mobile/lib/features/feed/domain/entities/post_entity.dart
class PostEntity {
  final String id;
  final String authorName;
  final String? authorHeadline;
  final String? authorAvatar;
  final String content;
  final String? postType;
  final String? imageUrl;
  final int likeCount;
  final int commentCount;
  final String createdAt;
  final bool isLiked;
  final bool isBookmarked;

  const PostEntity({
    required this.id,
    required this.authorName,
    this.authorHeadline,
    this.authorAvatar,
    required this.content,
    this.postType = 'text',
    this.imageUrl,
    this.likeCount = 0,
    this.commentCount = 0,
    required this.createdAt,
    this.isLiked = false,
    this.isBookmarked = false,
  });

  PostEntity copyWith({
    String? id,
    String? authorName,
    String? authorHeadline,
    String? authorAvatar,
    String? content,
    String? postType,
    String? imageUrl,
    int? likeCount,
    int? commentCount,
    String? createdAt,
    bool? isLiked,
    bool? isBookmarked,
  }) {
    return PostEntity(
      id: id ?? this.id,
      authorName: authorName ?? this.authorName,
      authorHeadline: authorHeadline ?? this.authorHeadline,
      authorAvatar: authorAvatar ?? this.authorAvatar,
      content: content ?? this.content,
      postType: postType ?? this.postType,
      imageUrl: imageUrl ?? this.imageUrl,
      likeCount: likeCount ?? this.likeCount,
      commentCount: commentCount ?? this.commentCount,
      createdAt: createdAt ?? this.createdAt,
      isLiked: isLiked ?? this.isLiked,
      isBookmarked: isBookmarked ?? this.isBookmarked,
    );
  }
}

File: backend/acadyk-api/src/main/kotlin/com/acadyk/modules/posts/entity/PostEntity.kt
package com.acadyk.modules.posts.entity

import com.acadyk.modules.profiles.entity.ProfileEntity
import jakarta.persistence.*
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "posts")
data class PostEntity(
    @Id
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "author_id", nullable = false)
    val author: ProfileEntity,

    @Column(columnDefinition = "TEXT", nullable = false)
    var content: String,

    var postType: String = "text",
    var visibility: String = "public",
    var imageUrl: String? = null,
    var likesCount: Int = 0,
    var commentsCount: Int = 0,
    var sharesCount: Int = 0,

    @Column(nullable = false)
    val createdAt: Instant = Instant.now(),

    @Column(nullable = false)
    var updatedAt: Instant = Instant.now(),

    var deletedAt: Instant? = null
)

@Entity
@Table(name = "post_media")
data class PostMediaEntity(
    @Id
    val id: UUID = UUID.randomUUID(),

    @Column(nullable = false)
    val postId: UUID,

    @Column(nullable = false)
    val mediaUrl: String,

    var mediaType: String = "image",
    var thumbnailUrl: String? = null,
    var position: Int = 0,
    val createdAt: Instant = Instant.now()
)
